addon/components/async-button.js

```diff
@@ -38,9 +38,18 @@ const ButtonComponent = Component.extend(positionalParamsMixin, {
       set(this, 'promise', promise);
     };
 
-    let actionArguments = ['action', callbackHandler, ...params];
+    if (typeof this.attrs.action === 'function') {
+      let promise = this.attrs.action();
```

Comment:
I would prefer not to provide a release that deprecates the old behavior before we break API.